### PR TITLE
Tendermint integration

### PR DESCRIPTION
**Pull request by pienkowb**

### CODE REVIEW

**Review by otaviopace — APPROVED** [review]

File: global/global.ts [hunk 1]
```diff
@@ -92,6 +100,53 @@ export enum TypeId {
...
+  TendermintArrayValidator = 124,
+  TendermintPublicKey = 125,
+  TendermintTxResult = 126,
+  TendermintResponseDeliverTx = 127,
+  TendermintEvent = 128,
+  TendermintEventAttribute = 129,
+  TendermintEventValidatorSetUpdates = 130,
+  TendermintDuration = 131,
+  TendermintTimestamp = 132,
+  TendermintEventData = 133,
```
Comment (leoyvens) [inline global/global.ts:149]: If we're going to start allocating index ranges for different chains, as we should, this will need to change.

Comment (otaviopace) [reply global/global.ts:149]: You mean having multiple enums for each `TypeId`?

Comment (leoyvens) [reply global/global.ts:149]: Not multiple enums, but ranges such as numbers 1000-2000 being reserved for tendermint.

Comment (otaviopace) [reply global/global.ts:149]: Oh I see, yeah that seems a good idea

**Pull request merged**